File: backend/app/schemas/ticket.py
"""Ticket schemas."""

from datetime import datetime
from pydantic import BaseModel, ConfigDict

from backend.app.models.ticket import TicketStatus


class TicketBase(BaseModel):
    """Base ticket schema."""
    title: str
    description: str | None = None
    asset_id: int | None = None


class TicketCreate(TicketBase):
    """Schema for creating a new ticket."""
    pass


class TicketUpdate(BaseModel):
    """Schema for updating an existing ticket."""
    title: str | None = None
    description: str | None = None
    status: TicketStatus | None = None
    asset_id: int | None = None


class TicketResponse(TicketBase):
    """Schema for ticket response."""
    id: int
    status: TicketStatus
    created_by_id: int
    approved_by_id: int | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TicketApprove(BaseModel):
    """Schema for approving a ticket."""
    approved: bool = True

File: frontend-react/src/types/ticket.ts
// 工单相关类型，对应后端 schemas/ticket.py 和 models/ticket.py。

// 工单的生命周期状态
export type TicketStatus =
  | 'draft'      // 草稿
  | 'submitted'  // 已提交，等待审批
  | 'approved'   // 已审批，可执行 ActionRun
  | 'running'    // 执行中
  | 'done'       // 完成
  | 'failed'     // 失败
  | 'closed'     // 已关闭

export interface Ticket {
  id: number
  title: string
  description: string | null
  status: TicketStatus
  asset_id: number | null
  created_by_id: number
  approved_by_id: number | null
  created_at: string
  updated_at: string
}

export interface TicketCreate {
  title: string
  description?: string
  asset_id?: number
}
